test/integration/generated/mediaTypes/src/mediaTypesClient.ts

```diff
@@ -27,26 +29,51 @@ class MediaTypesClient extends MediaTypesClientContext {
 
   /**
    * Analyze body, that could be different media types.
+   * @param contentType Upload file type
    * @param options The options parameters.
    */
   analyzeBody(
-    options?: MediaTypesClientAnalyzeBodyOptionalParams
+    contentType: ContentType,
+    options?: MediaTypesClientAnalyzeBody$binaryOptionalParams
+  ): Promise<MediaTypesClientAnalyzeBodyResponse>;
+  /**
+   * Analyze body, that could be different media types.
+   * @param contentType Body Parameter content-type
+   * @param options The options parameters.
+   */
+  analyzeBody(
```

Comment:
Oh, one thing I forgot: Why is the second overload needed?  Is `application/json` not covered by the `ContentType` enum?

Reply:
Correct. ContentType contains the content-types that are binary media types. e.g. application/pdf.
This actually works well for us because one overload accepts the binary content-types (and the request body would accept coreHttp.HttpRequestBody), and the other overload accepts just application/json (so the body parameter is a structured object).